persist/json.go

```diff
@@ -1,93 +1,238 @@
 package persist
 
 import (
+	"bytes"
 	"encoding/json"
-	"io"
+	"io/ioutil"
 	"os"
+	"strings"
+
+	"github.com/NebulousLabs/Sia/build"
+	"github.com/NebulousLabs/Sia/crypto"
 )
 
-// Load loads json data from a reader.
-func Load(meta Metadata, data interface{}, r io.Reader) error {
+// readJSON will try to read a persisted json object from a file.
+func readJSON(meta Metadata, object interface{}, filename string) error {
+	// Open the file.
+	file, err := os.Open(filename)
+	if os.IsNotExist(err) {
+		return err
+	}
+	if err != nil {
+		return build.ExtendErr("unable to open persisted json object file", err)
+	}
+	defer file.Close()
+
+	// Read the metadata from the file.
 	var header, version string
-	dec := json.NewDecoder(r)
+	dec := json.NewDecoder(file)
 	if err := dec.Decode(&header); err != nil {
-		return err
+		return build.ExtendErr("unable to read header from persisted json object file", err)
 	}
 	if header != meta.Header {
 		return ErrBadHeader
 	}
 	if err := dec.Decode(&version); err != nil {
-		return err
+		return build.ExtendErr("unable to read version from persisted json object file", err)
 	}
 	if version != meta.Version {
 		return ErrBadVersion
 	}
-	if err := dec.Decode(data); err != nil {
-		return err
+
+	// Read everything else.
+	remainingBytes, err := ioutil.ReadAll(dec.Buffered())
+	if err != nil {
+		return build.ExtendErr("unable to read persisted json object data", err)
```

Comment:
oic, it's so you can calculate the checksum. Could probably achieve the same using a `io.TeeReader` but this is fine